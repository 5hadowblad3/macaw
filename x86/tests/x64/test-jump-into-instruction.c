#include "util.h"

void _start() {
  long long x;
here:
  // This makes an instruction with a short 0x90 nop slide.
  x = 0x9090909090909090;
  // The +4 is completely ignored by gcc. You have to modify the assembly by
  // hand after compiling to make this jump into the middle of an emitted
  // instruction.
  goto *(&&here + 4);
  EXIT();
}
