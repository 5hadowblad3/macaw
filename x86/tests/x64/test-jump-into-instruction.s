	.file	"test-jump-into-instruction.c"
	.text
	.globl	_start
	.type	_start, @function
_start:
.LFB0:
	.cfi_startproc
	pushq	%rbp
	.cfi_def_cfa_offset 16
	.cfi_offset 6, -16
	movq	%rsp, %rbp
	.cfi_def_cfa_register 6
.L2:
	movabsq	$-8029759185026510704, %rax
	movq	%rax, -8(%rbp)
	nop
	jmp	.L2+4
	.cfi_endproc
.LFE0:
	.size	_start, .-_start
	.ident	"GCC: (GNU) 7.2.1 20170915 (Red Hat 7.2.1-2)"
	.section	.note.GNU-stack,"",@progbits
