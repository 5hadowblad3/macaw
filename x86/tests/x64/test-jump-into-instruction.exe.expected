R { funcs = [(Addr 0 0x400144,
              [(Addr 0 0x400144, 14), (Addr 0 0x40014c, 6), (Addr 0 0x400152, 7)])]
  , ignoreBlocks = []
  }
